app/src/pages/[resourceType]/[id].tsx

```diff
@@ -73,6 +75,20 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
   );
 }
 
+/**
+ * Function which extracts specified content from JSON data and then returns the decoded
+ * version of that content (in this case the ELM/CQL)
+ * @returns The decoded version of the ELM/CQL content
+ */
+function Decode(link: string, jsonData: FhirArtifact) {
+  const decodedLanguage = useMemo(() => {
+    if (jsonData.resourceType === 'Measure') return null;
+    const encodedLanguage = (jsonData as fhir4.Library).content?.find(e => e.contentType === link)?.data;
+    return encodedLanguage ? Buffer.from(encodedLanguage, 'base64').toString() : null;
+  }, [jsonData, link]);
+  return decodedLanguage;
```

Comment:
Suggest moving the `useMemo` part back to the component and just have this function do the extraction.